app/navigation/Navigator.scala

```diff
@@ -42,8 +43,8 @@ class Navigator @Inject() (categorisationService: CategorisationService) {
     case HasNiphlPage                              => navigateFromHasNiphl
     case NiphlNumberPage                           => _ => routes.CyaCreateProfileController.onPageLoad
     case UkimsNumberUpdatePage                     => _ => routes.CyaMaintainProfileController.onPageLoadUkimsNumber
-    case HasNirmsUpdatePage                        => navigateFromHasNirmsUpdate
-    case NirmsNumberUpdatePage                     => _ => routes.ProfileController.onPageLoad()
+    case HasNirmsUpdatePage                        => answers => navigateFromHasNirmsUpdate(answers, NormalMode)
```

Comment:
this feels wrong we should have two separate functions one for check mode and one for normal mode like the other things in this case list have